### DISCUSSION TITLE
Unable to create DB using MongoDB Compass

### DISCUSSION THREAD

**Discussion by PlatonB** [FerretDB/FerretDB#2625] [General]
### FerretDB version

version: v1.1.0
commit: 3344cbb98bb744dd044bcf2d51fe9ab65db22f0b
branch: unknown
dirty: true
package: deb
debugBuild: false

### Backend

psql (PostgreSQL) 14.7 (Ubuntu 14.7-0ubuntu0.22.04.1)

### Environment

- OS: elementary OS 7
- Deployment: DEB
- MongoDB Compass 1.36.4 (flatpak)

### What did you do?

1. Prepared PostgreSQL and FerretDB for use. Stopped mongod to avoid conflicts.
```
sudo apt install postgresql
sudo systemctl enable postgresql
sudo -u postgres createuser platon
sudo -u postgres createdb platon
```
```
sudo systemctl stop mongod
```
```
sudo apt install /path/to/ferretdb.deb
ferretdb
```

2. Connected via Compass without any errors.
![Compass_FerretDB_URI](https://github.com/FerretDB/FerretDB/assets/25541767/9ff12b40-3455-4971-a005-8f70bb6144b6)
![Compass_FerretDB_Databases](https://github.com/FerretDB/FerretDB/assets/25541767/67b08235-9c94-481c-a5d3-c400163c62d9)

3. Then I tried to create DB and table. An error appeared.
![Compass_FerretDB_pass_auth_error](https://github.com/FerretDB/FerretDB/assets/25541767/4bb2243a-a5d5-4abf-af1f-29e0b0488317)

### What did you expect to see?

Since FerretDB [works with Studio 3T](https://blog.ferretdb.io/using-ferretdb-with-studio-3t/), I expect it to work with Compass as well.

### What did you see instead?

Full error code printed by `ferretdb` running in the terminal:
```
2023-05-12T15:14:31.836+0300	WARN	pg/pg.go:137	DBPool: authentication failed	{"username": "", "error": "pgdb.checkConnection: failed to connect to `host=127.0.0.1 user=platon database=ferretdb`: failed SASL auth (FATAL: password authentication failed for user \"platon\" (SQLSTATE 28P01))"}
2023-05-12T15:14:31.836+0300	WARN	// 127.0.0.1:52620 -> 127.0.0.1:27017 	clientconn/conn.go:569	Response header: length:   347, id:    4, response_to:   23, opcode: OP_MSG
2023-05-12T15:14:31.836+0300	WARN	// 127.0.0.1:52620 -> 127.0.0.1:27017 	clientconn/conn.go:570	Response message:
{
  "Checksum": 0,
  "FlagBits": 0,
  "Sections": [
    {
      "Document": {
        "$k": [
          "ok",
          "errmsg",
          "code",
          "codeName"
        ],
        "ok": {
          "$f": 0
        },
        "errmsg": "[msg_create.go:37 pg.(*Handler).MsgCreate] [pg.go:138 pg.(*Handler).DBPool] pgdb.checkConnection: failed to connect to `host=127.0.0.1 user=platon database=ferretdb`: failed SASL auth (FATAL: password authentication failed for user \"platon\" (SQLSTATE 28P01))",
        "code": 1,
        "codeName": "InternalError"
      },
      "Kind": 0
    }
  ]
}
```

**Comment by AlekSi** [comment]
What credentials did you use in PostgreSQL and MongoDB URIs? See [here](https://docs.ferretdb.io/security/#authentication).

**Comment by PlatonB** [comment]
I used the URI from a [screenshot](https://blog.ferretdb.io/assets/images/image3-079546bcde3804e63eef8d214caa3ead.png) I found on the FerretDB blog. The [docs section](https://docs.ferretdb.io/security/#authentication) you referred to is difficult to understand.

**Comment by AlekSi** [comment]
Well, then, I guess your PostgreSQL installation requires credentials that are not provided. So it is not about MongoDB Compass at all.

See https://ubuntu.com/server/docs/databases-postgresql. The simplest solution for local testing could be enabling `trust` for all localhost connections; see https://www.postgresql.org/docs/current/auth-trust.html.

Or just use our all-in-one Docker image: https://github.com/FerretDB/FerretDB#quickstart

**Reply by PlatonB** [reply]
1. `sudo nano /etc/postgresql/14/main/pg_hba.conf`
2. `trust` instead of all values in the `METHOD` column.

Now another error appears:
```[msg_create.go:37 pg.(*Handler).MsgCreate] [pg.go:138 pg.(*Handler).DBPool] pgdb.checkConnection: failed to connect to `host=127.0.0.1 user=platon database=ferretdb`: server error (FATAL: database "ferretdb" does not exist (SQLSTATE 3D000))```

Full output:
```
2023-05-12T17:40:35.155+0300	WARN	pg/pg.go:137	DBPool: authentication failed	{"username": "", "error": "pgdb.checkConnection: failed to connect to `host=127.0.0.1 user=platon database=ferretdb`: server error (FATAL: database \"ferretdb\" does not exist (SQLSTATE 3D000))"}
2023-05-12T17:40:35.155+0300	WARN	// 127.0.0.1:50230 -> 127.0.0.1:27017 	clientconn/conn.go:569	Response header: length:   329, id:    4, response_to:   23, opcode: OP_MSG
2023-05-12T17:40:35.155+0300	WARN	// 127.0.0.1:50230 -> 127.0.0.1:27017 	clientconn/conn.go:570	Response message:
{
  "Checksum": 0,
  "FlagBits": 0,
  "Sections": [
    {
      "Document": {
        "$k": [
          "ok",
          "errmsg",
          "code",
          "codeName"
        ],
        "ok": {
          "$f": 0
        },
        "errmsg": "[msg_create.go:37 pg.(*Handler).MsgCreate] [pg.go:138 pg.(*Handler).DBPool] pgdb.checkConnection: failed to connect to `host=127.0.0.1 user=platon database=ferretdb`: server error (FATAL: database \"ferretdb\" does not exist (SQLSTATE 3D000))",
        "code": 1,
        "codeName": "InternalError"
      },
      "Kind": 0
    }
  ]
}
```

**Reply by AlekSi** [reply]
You did not provide PostgreSQL URI, so I could only guess that it ends with `/ferretdb` instead of `/platon` (that's a database you created).

**Reply by PlatonB** [reply]
I have now created a dummy DB named `ferretdb`:
```
sudo -u postgres createdb ferretdb
```

After that, trying to create a new DB via Compass leads to another error:
```
[msg_create.go:37 pg.(*Handler).MsgCreate] [pg.go:138 pg.(*Handler).DBPool] pgdb.checkConnection: "lc_collate" is "ru_RU.UTF-8"
```

**Reply by AlekSi** [reply]
We don't support that locale. See https://docs.ferretdb.io/understanding-ferretdb/#postgresql

**Reply by PlatonB** [reply]
After editing the locale in `/etc/postgresql/14/main/postgresql.conf` the following error displays:
```
[msg_create.go:37 pg.(*Handler).MsgCreate] [pg.go:138 pg.(*Handler).DBPool] pgdb.checkConnection: failed to connect to `host=127.0.0.1 user=platon database=ferretdb`: dial error (dial tcp 127.0.0.1:5432: connect: connection refused)
```

**Reply by AlekSi** [reply]
The most likely explanation is that PostgreSQL is not running or accepts no connections.

FWIW, you create a separate database with the given encoding and locate without editing PostgreSQL configuration files: https://www.postgresql.org/docs/current/sql-createdatabase.html

**Reply by PlatonB** [reply]
I recommend to write a tutorial similar to https://blog.ferretdb.io/using-ferretdb-with-studio-3t/, but related to Compass and PyMongo.

**Reply by AlekSi** [reply]
Maybe this will be more helpful: https://github.com/FerretDB/FerretDB#quickstart
